recipe_app/tests.py

```diff
@@ -0,0 +1,322 @@
+        )
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(Recipe.objects.get(id=1).icon, "🍕")
+
+    def test_modify_recipe_ingredients(self):
+        response = self.client.patch(
+            "/recipes/1/",
+            data={
+                "ingredients": [
+                    {"name": "pineapple"},
+                    {"name": "ham"},
+                ]
+            },
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()["ingredients"]), 2)
+        self.assertEqual(Recipe.objects.count(), 2)
+        self.assertEqual(Ingredient.objects.count(), 6)
+        self.assertEqual(
+            RecipeSerializer(Recipe.objects.get(id=1)).data, response.json()
+        )
+
+    def test_modify_recipe_invalid(self):
+        response = self.client.patch(
+            "/recipes/1/",
+            data={"no_exist": "hello"},
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Recipe.objects.count(), 2)
+        self.assertEqual(Ingredient.objects.count(), 7)
+        self.assertEqual(
+            RecipeSerializer(Recipe.objects.get(id=1)).data, response.json()
+        )
+
+        # Check that it didn't modify the original recipe
+        self.assertEqual(
+            RecipeSerializer(Recipe.objects.get(id=1)).data, self.recipes[0]
+        )
+
+    def test_modify_recipe_not_found(self):
+        response = self.client.patch(
+            "/recipes/100/",
+            data={"name": "Mac and Cheese"},
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(Recipe.objects.count(), 2)
+        self.assertEqual(Ingredient.objects.count(), 7)
+
+    def test_modify_recipe_no_id(self):
+        self.assertRaises(
+            TypeError,
+            self.client.patch,
+            "/recipes/",
+            data={"name": "Mac and Cheese"},
+            content_type="application/json",
+        )
+
+    # * Test cases for DELETE requests
+
+    def test_delete_recipe(self):
+        response = self.client.delete("/recipes/2/")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Recipe.objects.count(), 1)
+        self.assertEqual(Ingredient.objects.count(), 3)
+
+    def test_delete_recipe_not_found(self):
+        response = self.client.delete("/recipes/100/")
+
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(Recipe.objects.count(), 2)
+        self.assertEqual(Ingredient.objects.count(), 7)
+
+    def test_delete_recipe_no_id(self):
+        self.assertRaises(
+            TypeError,
+            self.client.delete,
+            "/recipes/",
+        )
+
+    # * Test relationships between Recipe and Ingredient
+
+    def test_one_recipe_to_one_ingredient(self):
```

Comment:
I don't think this belongs in `RecipeViewTestCase`, as it's more of a model test :)

Reply:
Agreed, moved to ModelTestCase